From 81693ba39970ec4fe5eca41686c8ef1efb9d087a Mon Sep 17 00:00:00 2001
From: Brenton Simpson <appsforartists@google.com>
Date: Tue, 30 Jan 2024 12:00:43 -0800
Subject: [PATCH] boot: add support for tablet hardware buttons (vol up/down,
 power)

Fixes #30598
---
 src/boot/efi/boot.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/boot/efi/boot.c b/src/boot/efi/boot.c
index 437e85e62d4b4..b71de3042ea1c 100644
--- a/src/boot/efi/boot.c
+++ b/src/boot/efi/boot.c
@@ -826,6 +826,7 @@ static bool menu_run(
 
                 switch (key) {
                 case KEYPRESS(0, SCAN_UP, 0):
+                case KEYPRESS(0, SCAN_VOLUME_UP, 0):
                 case KEYPRESS(0, 0, 'k'):
                 case KEYPRESS(0, 0, 'K'):
                         if (idx_highlight > 0)
@@ -833,6 +834,7 @@ static bool menu_run(
                         break;
 
                 case KEYPRESS(0, SCAN_DOWN, 0):
+                case KEYPRESS(0, SCAN_VOLUME_DOWN, 0):
                 case KEYPRESS(0, 0, 'j'):
                 case KEYPRESS(0, 0, 'J'):
                         if (idx_highlight < config->n_entries-1)
@@ -873,6 +875,7 @@ static bool menu_run(
                 case KEYPRESS(0, SCAN_F3, 0): /* EZpad Mini 4s firmware sends malformed events */
                 case KEYPRESS(0, SCAN_F3, '\r'): /* Teclast X98+ II firmware sends malformed events */
                 case KEYPRESS(0, SCAN_RIGHT, 0):
+                case KEYPRESS(0, SCAN_SUSPEND, 0):
                         exit = true;
                         break;
 
